base/networking_tests.c

```diff
@@ -105,6 +105,56 @@ Ensure (networking, validate_port_range)
   assert_that (validate_port_range ("   T  : 1"), is_equal_to (0));
 }
 
+Ensure (networking, port_range_ranges)
+{
+  /* Valid port ranges. U:,T: are empty ranges which are ignored. */
+  const gchar *valid_portrange1 = "1,10-13,10-10,T:1-2,U:10-12,U:,T:";
+  assert_that (validate_port_range (valid_portrange1), is_equal_to (0));
+
+  assert_that (port_range_ranges (NULL), is_null);
+  assert_that (port_range_ranges (valid_portrange1), is_not_null);
+
+  array_t *valid_portrange1_ranges = port_range_ranges (valid_portrange1);
```

Comment:
I'd much rather see the variables all declared at the top of the block (but keeping the definitions at the point of use).